### ISSUE TITLE
Negative loss in fit_surrogate_posterior

### ISSUE THREAD

**Issue by IvanUkhov** [tensorflow/probability#1232]
I use the following code for invoking optimization:

```python
def _infer(
    model: tp.sts.StructuralTimeSeries,
    y: pd.Series,
    learning_rate: float = 0.01,
    step_count: int = 1000,
    draw_count: int = 1000,
    random_state: int = 42,
) -> Tuple[Dict[str, tf.Tensor], Dict[str, tf.Tensor]]:
    optimizer = tf.optimizers.Adam(learning_rate=learning_rate)
    surrogate = tp.sts.build_factored_surrogate_posterior(model=model)

    @tf.function(experimental_compile=True)
    def _process():
        losses = tp.vi.fit_surrogate_posterior(
            target_log_prob_fn=model.joint_log_prob(
                observed_time_series=y),
            surrogate_posterior=surrogate,
            optimizer=optimizer,
            num_steps=step_count,
            seed=random_state,
        )
        draws = surrogate.sample(draw_count)
        return draws, dict(loss=losses)

    return _process()
```

When plotting the loss, I have noticed that it goes below zero:

<img width="805" alt="Screenshot 2021-02-07 at 13 10 48" src="https://user-images.githubusercontent.com/57440/107146066-0c4e0900-6946-11eb-9e0b-31eaadc20e12.png">

I am wondering if this is something expected from the default `variational_loss_fn`, which is a Monte Carlo approximation to the standard evidence lower bound. If so, what is the interpretation of this behavior? Thank you!

**Comment by davmre** [comment]
The loss approximates the negative evidence lower bound, so a negative loss means that the ELBO (and thus the log model evidence itself) is positive.

There's nothing particularly strange about this: if `p(x) > 1` (this cannot happen if `x` is discrete but can easily occur when `x` is continuous and `p(x)` is a density function), then we'll have positive log evidence---`log p(x) > 0`---and so potentially a positive ELBO.

**Comment by IvanUkhov** [comment]
Thank you for the quick response. That makes sense. The definition is as follows:

```math
L(X) = ln p(X) − KL[q(Z) || p(Z|X)].
```

For simplicity, assume a perfect fit:

```math
L(X) = ln p(X).
```

Here it is clear that the lower bound can have any sign.
